end-to-end-test/remote/specs/core/customTabs.spec.js

```diff
@@ -0,0 +1,397 @@
+
+describe('Patient Cohort View Custom Tab Tests', () => {
+    const conf = [
+        {
+            title: 'Sync Tab',
+            id: 'customTab1',
+            location: 'PATIENT_PAGE',
+            mountCallback: `(div)=>{
+                    $(div).html("tab for patient " + window.location.search.split("=").slice(-1))
+                }`,
+        },
+        {
+            title: 'Async Tab',
+            id: 'customTab2',
+            location: 'PATIENT_PAGE',
+            hideAsync: `()=>{
+                           return new Promise((resolve)=>{
+                                setTimeout(()=>{
+                                    resolve(true);
+                                }, 2000);
+                           });
+                    }`,
+        },
+    ];
+
+    it('Routing directly to conditional tab forces it to show', () => {
+        goToUrlWithCustomTabConfig(patientUrl, conf);
+
+        $('.mainTabs').waitForDisplayed();
+
+        // when we are directly routed to tab
+        // the navigation tab button DOES show
+        // even though hideAsync is still pending
+        assert.equal($('=Sync Tab').isDisplayed(), true, 'We see sync tab');
+
+        assert.equal(
+            $('=Async Tab').isDisplayed(),
+            false,
+            'We do NOT see async tab initially'
+        );
+
+        browser.pause(2000);
+
+        assert.equal(
+            $('=Async Tab').isDisplayed(),
+            true,
+            'Now we see async tab'
+        );
+
+        $('.nextPageBtn').click();
+
+        $('.mainTabs').waitForDisplayed();
+
+        assert.equal(
+            $('=Async Tab').isDisplayed(),
+            false,
+            'Async tab is pending again'
+        );
+
+        browser.pause(2000);
+
+        assert.equal(
+            $('=Async Tab').isDisplayed(),
+            true,
+            'Async tab is pending again'
+        );
+    });
+
+    it('Navigating between patients changes tab contents', () => {
+        goToUrlWithCustomTabConfig(patientUrl, conf);
+
+        $('.mainTabs').waitForDisplayed();
+
+        $('=Sync Tab').click();
+
+        // when we are directly routed to tab
+        // the navigation tab button DOES show
+        // even though hideAsync is still pending
+        assert.equal(
+            $('div=tab for patient TCGA-AP-A053').isDisplayed(),
+            true,
+            'We first patient'
+        );
+
+        $('.nextPageBtn').click();
+
+        $('.mainTabs').waitForDisplayed();
+
+        assert.equal(
+            $('div=tab for patient TCGA-BG-A0M8').isDisplayed(),
+            true,
+            'We see next patient'
+        );
+
+        $('.prevPageBtn').click();
+
+        assert.equal(
+            $('div=tab for patient TCGA-AP-A053').isDisplayed(),
+            true,
+            'We first patient'
+        );
+    });
+});
+
+runTests('ResultsView', resultsUrl, 'RESULTS_PAGE');
+
+runTests('PatientView', patientUrl, 'PATIENT_PAGE');
```

Comment:
I noticed that Comparison View is also supported not. Should we not include a test for it?